clients/src/main/java/org/apache/kafka/clients/consumer/internals/MembershipManagerImpl.java

```diff
@@ -673,6 +673,9 @@ public void onHeartbeatRequestSent() {
         MemberState state = state();
         if (isStaled()) {
             log.debug("Member {} is staled and is therefore leaving the group.  It will rejoin upon the next poll.", memberEpoch);
+            // TODO: Integrate partition revocation/loss callback
+            // Clear the current assignment and subscribed partitions because the member has left the group
+            updateSubscription(new TreeSet<>(TOPIC_ID_PARTITION_COMPARATOR), true);
```

Comment:
Should we rather call this before `transitionTo(MemberState.STALED);`?

Reply:
Do you mean in `transitionToStaled()`? That makes sense to me.

Reply:
> Do you mean in `transitionToStaled()`? That makes sense to me.

Yes.

Reply:
Thanks, I originally thought we wanted to abandon the partitions after sending the leave group.

Reply:
Is it correct that we have this call now in two places?